src/mono/mono/mini/mini-llvm.c

```diff
@@ -1610,7 +1610,11 @@ sig_to_llvm_sig_full (EmitContext *ctx, MonoMethodSignature *sig, LLVMCallInfo *
 	case LLVMArgVtypeAsScalar: {
 		int size = mono_class_value_size (mono_class_from_mono_type_internal (rtype), NULL);
 		/* LLVM models this by returning an int */
-		if (size < TARGET_SIZEOF_VOID_P) {
+		if (size == 0) {
+			/* Empty struct with LayoutKind attribute and without specified size */
+			g_assert(cinfo->ret.nslots == 0);
+			ret_type = LLVMIntType (8);
```

Comment:
> I don't think void return type would even enter this code path. Will verify.

I meant could you change it like this:

```suggestion
			ret_type = LLVMVoidType ();
```

Reply:
Yeah. I tried this initially and it resulted in ill formed llvm ir.